pkg/schedule/cron_schedule.go

```diff
@@ -0,0 +1,121 @@
+/*
+http://www.apache.org/licenses/LICENSE-2.0.txt
+
+Copyright 2016 Intel Corporation
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package schedule
+
+import (
+	"errors"
+	"time"
+
+	"github.com/robfig/cron"
+)
+
+var ErrMissingCronEntry = errors.New("Cron entry is missing")
+
+// CronSchedule is a schedule that waits as long as specified in cron entry
+type CronSchedule struct {
+	entry    string
+	state    ScheduleState
+	schedule *cron.Cron
+}
+
+// NewCronSchedule creates and starts new cron schedule and returns an instance of CronSchedule
+func NewCronSchedule(entry string) *CronSchedule {
+	schedule := cron.New()
+	schedule.Start()
+	return &CronSchedule{
+		entry:    entry,
+		schedule: schedule,
+	}
+}
+
+// GetState returns state of CronSchedule
+func (c *CronSchedule) GetState() ScheduleState {
+	return c.state
+}
+
+// Validate returns error if cron entry dosn't match crontab format
+func (c *CronSchedule) Validate() error {
+	if c.entry == "" {
+		return ErrMissingCronEntry
+	}
+	_, err := cron.Parse(c.entry)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+// Wait waits as long as specified in cron entry
+func (c *CronSchedule) Wait(last time.Time) Response {
+	// calculate misses
+	var misses uint
+	e := c.schedule.Entries()
+	if len(e) > 0 {
+		s := e[0].Schedule
+		now := time.Now()
+		for next := last; next.Before(now); {
+			next = s.Next(next)
+			if next.After(now) {
+				break
+			}
+			misses++
+		}
+	}
+
+	// wait as long as specified in cron entry
+	done := make(chan struct{})
+	err := c.schedule.AddFunc(c.entry, func() { done <- struct{}{} })
```

Comment:
Am I reading this wrong or does this add the same entry to the running scheduler for every call to Wait()?